src/rdkafka_request.c

```diff
@@ -4473,12 +4473,11 @@ rd_kafka_EndTxnRequest (rd_kafka_broker_t *rkb,
 
         /* Committed */
         rd_kafka_buf_write_bool(rkbuf, committed);
-
+        rkbuf->rkbuf_u.EndTxn.commit = committed;
 
         rd_kafka_buf_ApiVersion_set(rkbuf, ApiVersion, 0);
 
-        /* Let the handler perform retries */
-        rkbuf->rkbuf_max_retries = RD_KAFKA_REQUEST_NO_RETRIES;
+        rkbuf->rkbuf_max_retries = RD_KAFKA_REQUEST_MAX_RETRIES;
```

Comment:
a couple of other txn requests set this to 3, wondering if there's a good reason for that.

Reply:
I think they're all NO or MAX retries now, shouldn't be any 3:ers left, where you seeing them?